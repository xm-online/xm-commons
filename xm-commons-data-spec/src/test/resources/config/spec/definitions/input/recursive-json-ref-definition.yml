test-items:
    - key: store/GET-FULL-ADDRESS
      inputSpec: |
          {
            "type": "object",
            "properties": {
              "uaa": {
                "$ref": "#/xmDefinition/address"
              }
            }
          }

definitions:
    - key: address
      ref: json-definitions/address.json
    - key: geoAddress
      ref: json-definitions/geoAddress.json
