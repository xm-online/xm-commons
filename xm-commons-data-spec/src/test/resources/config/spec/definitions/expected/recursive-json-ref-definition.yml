test-items:
    - key: store/GET-FULL-ADDRESS
      inputSpec: |
          {
              "type":"object",
              "properties":{
                "uaa":{
                  "$ref":"#/xmDefinition/address"
                }
              },
              "xmDefinition":{
                "address":{
                  "type":"object",
                  "properties":{
                    "address":{
                      "type":"string"
                    },
                    "geoAddress":{
                      "$ref":"#/xmDefinition/geoAddress"
                    },
                    "formattedAddress":{
                      "type":"string"
                    }
                  }
                },
                "geoAddress":{
                  "type":"object",
                  "properties":{
                    "location":{
                      "lat":{
                        "type":"number"
                      },
                      "lng":{
                        "type":"number"
                      }
                    },
                    "place_id":{
                      "type":"string"
                    },
                    "utcOffset":{
                      "type":"integer"
                    },
                    "components":{
                      "type":"object"
                    }
                  }
                }
              }
            }

definitions:
    - key: address
      ref: json-definitions/address.json
    - key: geoAddress
      ref: json-definitions/geoAddress.json
